.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,37 @@
+name: CI Test Suite
+
+on: 
+    push:
+        branches: 
+          - main
+    pull_request: 
+        branches:
+          - main
+
+jobs:
+    test:
+        runs-on: ubuntu-latest
+
+        steps:
+            - name: Checkout Code
+              uses: actions/checkout@v4
+
+            - name: Setup go
+              uses: actions/setup-go@v5
+              with:
+                  go-version: 1.23
+
+            - name: Install Dependencies
+              run: go mod tidy
+            
+            - name: Run Tests
+              run: go test ./... -json | tee test-results.log
```

Comment:
You can redirect the contents of the `go test ...` command to a JSON file like this instead - `go test ./... -json > test-results.json`.